### PR TITLE
Include BPF FS mount status in sysdump output

### LINKED ISSUE #12057: Include BPF FS mount status in sysdump output
It would be helpful for later introspection if the status of BPF filesystem mounting was reported either through a new `cilium bpf fs show` command or directly in the `cilium status` output.

It should include the location of the BPF FS and whether Cilium specifically mounted the FS because the OS did not automatically provide it. (See the warning messages relating to BPFFS mounting that occur early in Daemon startup).

For testing this feature it may be worth considering to have an option to tweak the helm charts to omit the BPFFS mounting from the host.

### PR DESCRIPTION
**Pull request by soumynathan**
This PR adds in a new cilium bpf command to show the status of
the bpffs filesystem mount.

Fixes: #12057
Signed-off-by: Swaminathan Vasudevan <svasudevan@suse.com>

### CODE REVIEW

File: cilium/cmd/bpf_mountfs_show.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+			if mountInfo.FilesystemType == "bpf" {
+				mountPoint = mountInfo.MountPoint
+				mountfsType = mountInfo.FilesystemType
+				mountfsStatus = true
+				break
+			}
+		}
+	} else {
+		Fatalf("Unable to find the mount %s", err)
+	}
```
Comment (tklauser) [inline cilium/cmd/bpf_mountfs_show.go]: I'd suggest to use the common `if err != nil { }` pattern here which IMO is easier to read:

```suggestion
	if err != nil {
		Fatalf("Unable to find the bpf filesystem mount %s", err)
	}
	
	for _, mountInfo := range mountdetails {
		if mountInfo.FilesystemType == "bpf" {
			mountPoint = mountInfo.MountPoint
			mountfsType = mountInfo.FilesystemType
			mountfsStatus = true
			break
		}
	}
```

**Review by tklauser — CHANGES_REQUESTED** [review]

File: cilium/cmd/bpf_mountfs_show.go (hunk 1 above)
Comment (soumynathan) [reply cilium/cmd/bpf_mountfs_show.go]: Sure will address your comments.

File: cilium/cmd/bpf_mountfs_show.go [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+			mountPoint = mountInfo.MountPoint
+			mountfsType = mountInfo.FilesystemType
+			mountfsStatus = true
+			break
+		}
+	}
+	w := tabwriter.NewWriter(os.Stdout, 2, 0, 3, ' ', 0)
+	fmt.Fprintf(w, "Filesystem Mounted state:\t%t\n", mountfsStatus)
+	fmt.Fprintf(w, "Filesystem Mount point:\t%s\n", mountPoint)
+	fmt.Fprintf(w, "Filesystem Type:\t%s\n", mountfsType)
```
Comment (joestringer) [inline cilium/cmd/bpf_mountfs_show.go]: Even if there's just one line, I think it makes sense to print these in columns to be consistent with the other CLI formats.

File: cilium/cmd/bpf_mountfs_show.go [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+	Use:     "show",
+	Short:   "Show bpf filesystem mount details",
+	Example: "cilium bpf fs show",
+	Run: func(cmd *cobra.Command, args []string) {
+		getbpfmountFS(cmd, args)
+	},
+}
+
+func init() {
+	bpfmountFSCmd.AddCommand(bpfmountfsShowCmd)
```
Comment (joestringer) [inline cilium/cmd/bpf_mountfs_show.go:38]: Can we add JSON support too? For examples, see how the other CLI commands do this; it should be pretty simple.

File: cilium/cmd/bpf_mountfs_show.go [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+	"github.com/cilium/cilium/pkg/mountinfo"
+	"github.com/spf13/cobra"
+)
+
+// bpfmountfsShowCmd represents the bpf file system show command
+var bpfmountfsShowCmd = &cobra.Command{
+	Use:     "show",
+	Short:   "Show bpf filesystem mount details",
+	Example: "cilium bpf fs show",
```
Comment (joestringer) [inline cilium/cmd/bpf_mountfs_show.go:31]: Please add this new command to the bugtool (`bugtool/cmd/configuration.go`) so that sysdump will pick it up.

**Review by joestringer — CHANGES_REQUESTED** [review]
Looks good overall thanks, some minor suggestions below.

File: cilium/cmd/bpf_mountfs_show.go (hunk 4 above)
Comment (soumynathan) [reply cilium/cmd/bpf_mountfs_show.go:31]: Sure will add it to the bugtool.

File: cilium/cmd/bpf_mountfs_show.go (hunk 3 above)
Comment (soumynathan) [reply cilium/cmd/bpf_mountfs_show.go:38]: Sure will add it.

File: cilium/cmd/bpf_mountfs_show.go (hunk 2 above)
Comment (soumynathan) [reply cilium/cmd/bpf_mountfs_show.go]: Ok will check it out.

File: cilium/cmd/bpf_mountfs_show.go (hunk 1 above)
Comment (soumynathan) [reply cilium/cmd/bpf_mountfs_show.go]: Done

File: cilium/cmd/bpf_mountfs_show.go (hunk 4 above)
Comment (soumynathan) [reply cilium/cmd/bpf_mountfs_show.go:31]: Done

File: cilium/cmd/bpf_mountfs_show.go (hunk 3 above)
Comment (soumynathan) [reply cilium/cmd/bpf_mountfs_show.go:38]: Done.

File: cilium/cmd/bpf_mountfs_show.go (hunk 2 above)
Comment (soumynathan) [reply cilium/cmd/bpf_mountfs_show.go]: Done

**Review by tklauser — APPROVED** [review]

**Comment by soumynathan** [comment]
@joestringer Is there anything else that we need to do in this PR.

**Review by joestringer — APPROVED** [review]

**Pull request merged**